File: benchmarks/pb_scopes2/mono.py
def f_outer(n):
    inc = 2

    def inner(x):
        def inner_2(x):
            return x + inc
        return inner_2(x)

    tot = 0
    for i in xrange(n):
        tot = inner(tot)

    expect = 2 * n
    assert tot == expect

def run_iter(n):
    f_outer(n)

File: benchmarks/pb_scopes2/mono.php
<?php

function f_outer($n) {
	$inc = 2;

	$inner = function($x) use(&$inc) {
        $inner2 = function($x) use (&$inc) {
            return $x + $inc;
        };
        return $inner2($x);
	};

	$tot = 0;
	for ($i = 0; $i < $n; $i++) {
		$tot = $inner($tot);
	}

	$expect = 2 * $n;
	assert($tot == $expect);

}

function run_iter($n) {
	f_outer($n);
}

?>
